### PR TITLE
Add a feature which enables to link between notes

### PR DESCRIPTION
**Pull request by asmsuechan**
Hi. I implemented a feature which is link between notes.

c83f3d8be4c391c20f72fbcf631b6c9f

### CODE REVIEW

File: browser/main/NoteList/index.js [hunk 1]
```diff
@@ -163,6 +166,32 @@ class NoteList extends React.Component {
     ee.emit('list:moved')
   }
 
+  jumpNoteByHashHandler (event, noteHash) {
```
Comment (asmsuechan) [inline browser/main/NoteList/index.js]: I'd like to use this function for DRY.

**Comment by asmsuechan** [comment]
This is bad layout for snippet.
screen shot 2017-04-07 at 13 48 53

**Comment by asmsuechan** [comment]
Where should I put the link of the note to avoid that layout corruption in the snippet? I write my ideas. Any ideas?

## Slide the hash in snippet
Just I slide the `div` 240px. But actually, it's a little complicated because of this override https://github.com/BoostIO/Boostnote/blob/master/browser/main/Detail/SnippetNoteDetail.js#L583:L605.

screen shot 2017-04-07 at 14 13 11

## Share button
This button has a lot of scalability e.g. `Share in Evernote`, `Share in Slack`, etc.

screen shot 2017-04-07 at 14 10 14

I want not to implement it in this PR. Because I just want to add the feature of `link` for jumping note to note.

File: browser/components/MarkdownPreview.js [hunk 2]
```diff
@@ -255,6 +261,10 @@ export default class MarkdownPreview extends React.Component {
       el.addEventListener('click', this.checkboxClickHandler)
     })
 
+    _.forEach(this.refs.root.contentWindow.document.querySelectorAll('a'), (el) => {
+      el.addEventListener('click', this.linkClickHandler)
```
Comment (sota1235) [inline browser/components/MarkdownPreview.js]: You can write this line at [this line](https://github.com/BoostIO/Boostnote/pull/336/files#diff-a54dec67725b3e604a34d39a7fcb72f6R257).

File: browser/components/MarkdownPreview.js [hunk 3]
```diff
@@ -338,6 +348,14 @@ export default class MarkdownPreview extends React.Component {
     e.stopPropagation()
   }
 
+  handlelinkClick (e) {
+    const noteHash = e.target.hash
+    const regexIsNoteLink = /^#(.{20})-(.{20})$/
```
Comment (sota1235) [inline browser/components/MarkdownPreview.js]: Can you cut out this logic into the module?
If you do it, you can write unit test for the logic. (And I think this is business logic.)

File: browser/components/MarkdownPreview.js [hunk 4]
```diff
@@ -338,6 +348,14 @@ export default class MarkdownPreview extends React.Component {
...
+    if (regexIsNoteLink.test(noteHash)) {
+      eventEmitter.emit('list:jump', noteHash.replace(/^#/, ''))
```
Comment (sota1235) [inline browser/components/MarkdownPreview.js]: [Question]Don't you need to exec `e.preventDefault()` after emit an event?

File: browser/main/NoteList/index.js (hunk 1 above)
Comment (sota1235) [inline browser/main/NoteList/index.js]: You can do it by cut out this logic to the module.
Like this.

```javascript
export function jumpToNote(event, noteHash, router, location) {
  // brabra
}
```

You can use it on any other components.

**Comment by asmsuechan** [comment]
Close this PR because one of the specifications is changed. I'm going to create a `share` button mentioned in https://github.com/BoostIO/Boostnote/pull/336#issuecomment-292655545.

**Pull request closed without merging**